ahcore/utils/vector_database/user_manager.py

```diff
@@ -0,0 +1,123 @@
+from functools import wraps
+from logging import getLogger
+from typing import Any, Callable, TypeVar, cast
+
+from pymilvus import Collection, MilvusException, Role, connections, utility
+
+logger = getLogger(__name__)
+
+
+T = TypeVar("T", bound=Callable[..., Any])
+
+
+def handle_milvus_error(func: T) -> T:
+    """Decorator to handle exceptions raised by Milvus operations."""
+
+    @wraps(func)
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        try:
+            return func(*args, **kwargs)
+        except MilvusException as e:
+            logger.error(f"Milvus error in {func.__name__}: {e}")
+            return None
+
+    return cast(T, wrapper)
+
```

Comment:
Maybe we can do this more broadly as well handle_ahcore_error.

Reply:
Possibly, although here I have done it mostly to avoid entire runs from stopping when some vectors are not successfully pushed to the database.